src/domain/disjointDomain.ml

```diff
@@ -184,14 +184,24 @@ sig
   val may_be_equal: elt -> elt -> bool
 end
 
-module ProjectiveSetPairwiseMeet (E: Printable.S) (B: MayEqualSetDomain with type elt = E.t) (R: Representative with type elt = E.t): SetDomain.S with type elt = E.t = struct
+module ProjectiveSetPairwiseMeet (E: Lattice.S) (B: MayEqualSetDomain with type elt = E.t) (R: Representative with type elt = E.t): SetDomain.S with type elt = E.t = struct
   include ProjectiveSet (E) (B) (R)
 
   let meet m1 m2 =
     let meet_buckets b1 b2 acc =
       B.fold (fun e1 acc ->
           B.fold (fun e2 acc ->
-              if B.may_be_equal e1 e2 then
+              (* If they have the same representative, we use the normal meet within this bucket *)
+              if R.equal (R.of_elt e1) (R.of_elt e2) then
+                try
+                  let m = E.meet e1 e2 in
+                  if not (E.is_bot m) then
+                    add m acc
+                  else
+                    acc
+                with Lattice.Uncomparable ->
+                  failwith (GobPretty.sprintf "Elements %a and %a are in same bucket, but meet throws!" E.pretty e1 E.pretty e2)
```

Comment:
Why rethrow a different exception at all. If that happens then `Uncomparable` could also just propagate to top level and crash. The stacktrace would go through there but also include the inner stacktrace that gets dropped by raising a new exception.
It's truly exceptional (and never supposed to happen), so the user won't be able to do anything with the text anyway.

Reply:
This will point immediately at the problem and provide an explanation about what exact invariant is broken. I would consider this much more helpful than a stack trace that just says that some things were incomparable.

It's not a hill I am willing to die on though.